argus/introspection/cloud/windows.py

```diff
@@ -286,12 +299,21 @@ def get_network_interfaces(self):
             if len(details) < 6:
                 continue
 
+            # Must follow `argus.util.NETWORK_KEYS` model.
+            addresses, addresses6 = _getip_v4_v6(details[1])
+            gateways, gateways6 = _getip_v4_v6(details[2])
+            netmasks, netmasks6 = _getip_v4_v6(details[3])
+            dnses, dnses6 = _getip_v4_v6(details[4])
             nic = {
                 "mac": details[0],
-                "address": details[1].split(" ")[0],
-                "gateway": details[2].split(" ")[0],
-                "netmask": details[3].split(" ")[0],
-                "dns": details[4].split(" "),
+                "address": addresses[0] if addresses else None,
```

Comment:
Why don't you set them conditionally?

if V6: set address6 & friends
else: set address4 & friends.